src/App.tsx

```diff
@@ -1,35 +1,16 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
 
-function App() {
-  const [count, setCount] = useState(0)
+import SignupPage from "./pages/SignupPage";
 
+function App() {
   return (
-    <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<Navigate to="/signup" />} />
+        <Route path="/signup" element={<SignupPage />} />
+      </Routes>
+    </BrowserRouter>
```

Comment:
Most probably you will add more routes in here, and the code can get a little repetitive. However you could create an array of routes and map through it. Something like this:

```suggestion

// before the return: 
//const routes = [
// {path:'/', element: <Navigate to='/signup'/>},
// {path:'/signup' element:<SignupPage/>}];

    <BrowserRouter>
      <Routes>
      {routes.map(({path,element}) => <Route path={path} element={element}/>)}
      </Routes>
    </BrowserRouter>
```

Reply:
Another suggestion is to use constants for all the strings you type manually across the whole app.
For example here you typed twice `'/signup'`, but if you have a file of constants and you import it from there,  typing mistakes wont break the app, and if in the future the client asks to change it to "create account" you just change it in one place